pkg/protocol/http1/resp/response_test.go

```diff
@@ -489,7 +493,7 @@ func testResponseReadWithoutBody(t *testing.T, resp *protocol.Response, s string
 	if len(resp.Body()) != 0 {
 		t.Fatalf("Unexpected response body %q. Expected %q. response=%q", resp.Body(), "", s)
 	}
-	verifyResponseHeader(t, &resp.Header, expectedStatusCode, expectedContentLength, expectedContentType)
+	verifyResponseHeader(t, &resp.Header, expectedStatusCode, expectedContentLength, expectedContentType, "")
```

Comment:
Maybe add some tests that `expectedContentEncoding` is not equal to ""? 
Since func `verifyResponseHeader` now has the ability to verify `ContentEncoding`.